### PR TITLE
opengl - improve timing and add Windows DwmFlush support

### PR DESCRIPTION
**Pull request by avih**

### CODE REVIEW

**Comment by wm4** [comment]
I don't remember the discussions around it, but why is the option off by default?

**Comment by avih** [comment]
TL;DR: it's hard to detect when `DwmFlush()` would be beneficial and when it would not, mostly due to unknown driver behavior and lack of adequate Windows APIs, and guessing is not great by default IMO.

There are two levels which we need to detect in order to be able to use this automatically:

1. If the DWM is currently disabled (XP - always and also the DLL should be unavailable, Windows 7 - depending on theme/config, Windows 8 - practically never) then we need to disable the DWM timing, which means tracking the state of DWM (since it can change in runtime on Windows 7) - not hard, but we'd need more code and event handling. Detecting the DWM state only once on init is also an option but less robust.

2. Even if we know that DWM is enabled on this system right now, it seems that we can't know if the video window is composited or not, since especially on full screen the driver may or may not bypass the DWM and enter exclusive mode, and as far as we've looked (with @rossy), we couldn't find an API to tell us that. Though it seems that on both our systems full screen on the main monitor was NOT composited (but full screen on a secondary monitor does appear to be composited). On both our systems we were composited when windowed.

That being said, maybe it's reasonable to check once on init if DWM is enabled, and if yes then enable `dwmflush` on windowed but not in full screen (so `:dwmflush=1` by default), but again, it's a guess which could get it wrong, which is why I chose the "easy" approach of yet another option (TM).

**Comment by wm4** [comment]
Merged (squashed the LoadLibrary commit).
